taskcat/_config_types.py

```diff
@@ -109,12 +109,39 @@ def from_dict(cls, raw_test: dict, project_root="./"):
         return Test(**raw_test)
 
 
-class S3BucketConfig(str):
-    def __init__(self, public: bool = False, auto: bool = False):
-        self.region = ""
-        self.public = public
-        self.auto = auto
-        self.max_name_len = 63
-        self.name = self.__str__()
-        self.tags: list = []
-        super().__init__()
+class S3BucketConfig:
+    def __init__(self):
+        self.name = ""
+        self.public = False
+
+
+class AWSRegionObject:
+    def __init__(self, region_name: str, client_factory: ClientFactory):
+        self.name: str = region_name
+        self.s3bucket = None
+        self.account: Optional[str] = None
+        self._cf: ClientFactory = client_factory
+        self._credset_name: Optional[str] = None
+        self._credset_modify = True
+
+    @property
+    def credset_name(self) -> Optional[str]:
+        return self._credset_name
+
+    @credset_name.setter
+    def credset_name(self, value: str) -> None:
+        if not self._credset_modify:
+            return
```

Comment:
should this just silently do nothing? Might be better to raise an exception here.

Reply:
Yes. The intention is to iterate through a list of regions, set the credential_set appropriately, then "lock" the changes. Example:

https://github.com/taskcat/taskcat-v9/blob/9a57ec02961019d08d1e773a5ecc9078b46bfe62/taskcat/_config.py#L161-L180